packages/auth/src/core/credentials/email.ts

```diff
@@ -111,13 +113,48 @@ export class EmailAuthCredential extends AuthCredential {
 
   /** @internal */
   async _getIdTokenResponse(auth: AuthInternal): Promise<IdTokenResponse> {
+    async function internalSignInWithPassword(cred: EmailAuthCredential, withRecaptcha: boolean = false): Promise<IdTokenResponse> {
+      let request: SignInWithPasswordRequest;
+      if (withRecaptcha) {
+        const verifier = new RecaptchaEnterpriseVerifier(auth);
+        const captchaResponse = await verifier.verify('signInWithEmailPassword')
+        .catch(() => {
+          return verifier.verify('signInWithEmailPassword', true);
+        }).catch((error) => {
+          return Promise.reject(error);
+        });
+        request = {
+          returnSecureToken: true,
+          email: cred._email,
+          password: cred._password,
+          captchaResponse,
+          clientType: RecaptchaClientType.WEB,
+          recaptchaVersion: RecaptchaVersion.ENTERPRISE,
+        };
+      } else {
+        request = {
```

Comment:
This part of the request is shared with both branches of the if statement. Please move the base assignment outside of the if statement (i.e. to the top), and then remove the else branch.

In the recaptcha case, just assign the new fields.

That should save some bytes